app/javascript/components/widgets/components/widget/widget.js

```diff
@@ -53,7 +52,9 @@ const mapStateToProps = ({ widgets, location }, ownProps) => {
   const locationPath = `dashboards/${type || 'global'}/${locationUrl}`;
   const widgetQuery = `widget=${widget}`;
   const widgetState =
-    query && query[widget] ? `&${widget}=${query[widget]}` : '';
+    query && query[widget]
+      ? `&${widget}=${btoa(JSON.stringify(query[widget]))}`
```

Comment:
Ah ha! You should no longer need to do this. You can now find this state as a clean and tidy object on the `location.query.widgets` reducer in the store. You can use a selector to merge the props just like in the map!

Reply:
I don't do it to update the current url. Actually it's only for generate the share url that we need to pass to the `EXPLORE ON GFW` button on the embed view.